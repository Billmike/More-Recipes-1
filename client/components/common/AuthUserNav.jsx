import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

/**
 * User navigation when unathenticated
 *
 * @param {any} props 
 * @returns {object} React element
 */
const AuthUserNav = (props) => {
  const { handleLogoutUser } = props;
  return (
    <ul id="user-control" className="dropdown-content">
      <li>
        <a href="profile.html" className="black-text">
          Profile
          <i className="material-icons left">person</i>
        </a>
      </li>
      <li>
        <Link to="/user/recipes" className="black-text">
          My Recipes<i className="material-icons left">folder</i>
        </Link>
      </li>
      <li>
        <a
          href="recipes.html?#user-favorites"
          className="black-text"
        >
          My Favorites
          <i className="material-icons left">folder</i>
        </a>
      </li>
      <li className="divider" />
      <li>
        <a
          role="button"
          tabIndex="0"
          className="black-text"
          onClick={handleLogoutUser}
        >
          Logout
          <i className="material-icons left">lock</i>
        </a>
      </li>
    </ul>
  );
};

AuthUserNav.propTypes = {
  handleLogoutUser: PropTypes.func.isRequired,
};

export default AuthUserNav;
