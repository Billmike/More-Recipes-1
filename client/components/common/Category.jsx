import React from 'react';

/**
 * Category options component
 *
 * @returns {object} React element
 */
const Category = () => (
  <ul id="category" className="dropdown-content">
    <li><a href="#!" className="collection-item black-text">Breakfast</a></li>
    <li><a href="#!" className="collection-item black-text">Lunch</a></li>
    <li><a href="#!" className="collection-item black-text">Dinner</a></li>
    <li><a href="#!" className="collection-item black-text">Appetizer</a></li>
    <li><a href="#!" className="collection-item black-text">Main</a></li>
    <li><a href="#!" className="collection-item black-text">Dessert</a></li>
  </ul>
);

export default Category;
