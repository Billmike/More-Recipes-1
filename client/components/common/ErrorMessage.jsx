import React from 'react';
import PropTypes from 'prop-types';

const ErrorMessage = (props) => {
  const { message } = props;
  return (
    <div className="red-text center-align">
      {message}
    </div>
  );
};

ErrorMessage.propTypes = {
  message: PropTypes.string.isRequired,
};

export default ErrorMessage;
