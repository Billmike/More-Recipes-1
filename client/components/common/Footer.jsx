import React from 'react';
import { Link } from 'react-router-dom';

/**
 * Footer component
 *
 * @returns {object} React element
 */
const Footer = () => (
  <div className="page-footer deep-orange darken-1">
    <div className="container center-align">
      <div className="row">
        <div className="connect-icons col s12 m6 l6">
          <h5 className="white-text">Stay Connected</h5>
          <i className="fa fa-facebook-square fa-2x" />
          <i className="fa fa-twitter-square fa-2x" />
          <i className="fa fa-pinterest-square fa-2x" />
          <i className="fa fa-instagram fa-2x" />
        </div>
        <div className="col s12 m6 l4 offset-l2">
          <h5 className="white-text">Navigate</h5>
          <ul>
            <li>
              <Link to="/" className="grey-text text-lighten-3">Home</Link>
            </li>
            <li>
              <a className="grey-text text-lighten-3" href="#!">About Us</a>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div className="footer-copyright">
      <div className="container">
        Â© 2017 All rights reserved
      </div>
    </div>
  </div>
);

export default Footer;
