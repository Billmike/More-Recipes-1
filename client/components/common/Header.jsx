import React from 'react';

import Navbar from './Navbar';
import SideNav from './SideNav';

const Header = props => (
  <div>
    <div className="navbar-fixed">
      <nav className="deep-orange darken-4">
        <Navbar {...props} />
      </nav>
    </div>
    <SideNav />
  </div>
);

export default Header;
