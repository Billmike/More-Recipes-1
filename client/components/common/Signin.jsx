import React from 'react';
import PropTypes from 'prop-types';
import Dialog from 'material-ui/Dialog';
import FlatButton from 'material-ui/FlatButton';
import { Field, reduxForm } from 'redux-form';
import isEmpty from 'lodash/isEmpty';

import { required, isEmptyField } from '../../utils/validate';
import ErrorMessage from '../common/ErrorMessage';
import RenderField from '../common/RenderField';
import Spinner from '../common/Spinner';

const propTypes = {
  openSignin: PropTypes.bool.isRequired,
  handleToggleSigninModal: PropTypes.func.isRequired,
  handleToggleModal: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  handleSubmitSignin: PropTypes.func.isRequired,
  user: PropTypes.shape({
    isLoading: PropTypes.bool.isRequired,
    error: PropTypes.shape({
      message: PropTypes.string
    })
  }).isRequired,
};

/**
 * Signin component
 *
 * @param {any} props The props passed to component
 *
 * @returns {object} React element
 */
const Signin = (props) => {
  const {
    openSignin, handleSubmitSignin, handleToggleSigninModal,
    handleToggleModal, user: { isLoading, error }
  } = props;
  const { message } = error;

  const actions = [
    <FlatButton
      label="Cancel"
      secondary
      onClick={handleToggleSigninModal}
    />
  ];

  return (
    <div>
      <Dialog
        title="Login into your account"
        actions={actions}
        modal
        open={openSignin}
        autoScrollBodyContent
      >
        <div className="row">
          <form
            className="col s12"
            onSubmit={props.handleSubmit(handleSubmitSignin)}
          >
            <div className="row">
              <div className="input-field col s12">
                <Field
                  name="username"
                  label="Username"
                  component={RenderField}
                  type="text"
                  validate={[required, isEmptyField]}
                />
              </div>
            </div>
            <div className="row">
              <div className="input-field col s12">
                <Field
                  name="password"
                  label="Password"
                  component={RenderField}
                  type="password"
                  validate={required}
                />
              </div>
            </div>
            {!isEmpty(error) && <ErrorMessage message={message} />}
            <div className="row" />
            <div className="center-align">
              {isLoading && <Spinner size="small" />}
            </div>
            <div className="row center-align">
              <button
                type="submit"
                className={`btn btn-large waves-effect waves-light
                  indigo accent-2`}
                disabled={isLoading}
              >
                Sign In
              </button>
            </div>
            <div className="row center-align">
              Don&#39;t have an account? <a
                role="button"
                tabIndex="0"
                className="cursor-pointer blue-text"
                onClick={handleToggleModal}
              >
                Register
              </a>
            </div>
          </form>
        </div>
      </Dialog>
    </div>
  );
};

Signin.propTypes = propTypes;

export default reduxForm({
  form: 'Signin'
})(Signin);
