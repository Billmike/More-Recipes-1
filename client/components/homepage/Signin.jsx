import React from 'react';
import PropTypes from 'prop-types';
import Dialog from 'material-ui/Dialog';
import FlatButton from 'material-ui/FlatButton';
import { Field, reduxForm } from 'redux-form';
import isEmpty from 'lodash/isEmpty';

import { required, isEmptyField } from '../../utils/validate';
import ErrorMessage from '../common/ErrorMessage';

const customContentStyle = {
  width: '35%',
  maxWidth: '35%',
};

/**
 * Function to render each of the input fields
 *
 * @param {object} Field - The input field
 * @param {object} Field.input - The input field element
 * @param {string} Field.label - The input field label
 * @param {string} Field.type - The input field type
 * @param {object} Field.meta
 * @param {boolean} Field.meta.touched - The field input state
 * @param {string} Field.meta.error - Validation error message
 * @returns {object} Input element
 */
const renderField = ({
  input,
  label,
  type,
  meta: { touched, error }
}) => (
  <div>
    <label htmlFor={label}>{label}</label>
    <br />
    <div>
      <input
        {...input}
        type={type}
        placeholder={label}
      />
      {touched && (error && <span className="red-text">{error}</span>)}
    </div>
  </div>
);

/**
 * Signin react component
 *
 * @param {any} props The props passed to component
 * @returns {object} React element
 */
const Signin = (props) => {
  const { user: { error } } = props,
    { message } = error;

  const actions = [
    <FlatButton
      label="Cancel"
      secondary
      onClick={props.handleClose}
    />
  ];

  return (
    <div>
      <Dialog
        title="Login into your account"
        actions={actions}
        modal
        contentStyle={customContentStyle}
        open={props.open}
        autoScrollBodyContent
      >
        <div className="row">
          <form
            className="col s12"
            onSubmit={props.handleSubmit(props.onSubmit)}
          >
            <div className="row">
              <div className="input-field col s12">
                <Field
                  name="username"
                  label="Username"
                  component={renderField}
                  type="text"
                  validate={[required, isEmptyField]}
                />
              </div>
            </div>
            <div className="row">
              <div className="input-field col s12">
                <Field
                  name="password"
                  label="Password"
                  component={renderField}
                  type="password"
                  validate={required}
                />
              </div>
            </div>
            {!isEmpty(error) && <ErrorMessage message={message} /> }
            <div className="row" />
            <div className="row center-align">
              <button
                type="submit"
                className={`btn btn-large waves-effect waves-light
                  indigo accent-2`}
                disabled={props.submitting}
              >
                Sign In
              </button>
            </div>
          </form>
        </div>
      </Dialog>
    </div>
  );
};

// Signin props validation
Signin.propTypes = {
  open: PropTypes.bool.isRequired,
  handleClose: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  submitting: PropTypes.bool.isRequired,
  user: PropTypes.shape({
    error: PropTypes.shape({
      message: PropTypes.string
    })
  }).isRequired,
};

// renderfield props validation
renderField.propTypes = {
  input: PropTypes.shape().isRequired,
  label: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  meta: PropTypes.shape({
    touched: PropTypes.bool,
    error: PropTypes.string,
  }).isRequired,
};

export default reduxForm({
  form: 'Signin'
})(Signin);
