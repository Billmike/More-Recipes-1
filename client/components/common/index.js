import Header from './Header';
import Footer from './Footer';
import RecipeImage from './RecipeImage';

export default {
  Header,
  Footer,
  RecipeImage,
};
