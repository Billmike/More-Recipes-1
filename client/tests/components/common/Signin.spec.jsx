import React from 'react';

import { Signin } from '../../../components/common/Signin';

describe('Signin component', () => {
  const props = {
    openSignin: true,
    handleToggleSigninModal: jest.fn(),
    handleToggleModal: jest.fn(),
    handleSubmit: jest.fn(),
    handleSubmitSignin: jest.fn(),
    user: {
      isLoading: false,
      error: {},
    },
  };
  it('should render component', () => {
    const wrapper = shallow(<Signin {...props} />);

    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
  it('should show spinner when submitting details', () => {
    const newProps = {
      ...props,
      user: {
        ...props.user,
        isLoading: true,
        error: {},
      },
    };
    const wrapper = shallow(<Signin {...newProps} />);

    expect(wrapper).toBeDefined();
    expect(wrapper.find('Spinner').exists()).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });
  it('should show error in signin', () => {
    const newProps = {
      ...props,
      user: {
        ...props.user,
        isLoading: false,
        error: {
          message: 'Error in signin',
        },
      },
    };
    const wrapper = shallow(<Signin {...newProps} />);

    expect(wrapper).toBeDefined();
    expect(wrapper.find('ErrorMessage').exists()).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });
});
