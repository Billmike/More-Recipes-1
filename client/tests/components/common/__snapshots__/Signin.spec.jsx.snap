// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signin component should render component 1`] = `
<div>
  <Dialog
    actions={
      Array [
        <FlatButton
          disabled={false}
          fullWidth={false}
          label="Cancel"
          labelPosition="after"
          labelStyle={Object {}}
          onClick={[MockFunction]}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          primary={false}
          secondary={true}
        />,
      ]
    }
    autoDetectWindowHeight={true}
    autoScrollBodyContent={true}
    modal={true}
    open={true}
    repositionOnUpdate={true}
    title="Login into your account"
  >
    <div
      className="row"
    >
      <form
        className="col s12"
      >
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <Field
              component={[Function]}
              label="Username"
              name="username"
              type="text"
              validate={
                Array [
                  [Function],
                  [Function],
                ]
              }
            />
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <Field
              component={[Function]}
              label="Password"
              name="password"
              type="password"
              validate={[Function]}
            />
          </div>
        </div>
        <div
          className="row"
        />
        <div
          className="center-align"
        />
        <div
          className="row center-align"
        >
          <button
            className="btn btn-large waves-effect waves-light
                  indigo accent-2"
            disabled={false}
            id="signin"
            type="submit"
          >
            Sign In
          </button>
        </div>
        <div
          className="row center-align"
        >
          Don't have an account? 
          <a
            className="cursor-pointer blue-text"
            onClick={[MockFunction]}
            role="button"
            tabIndex="0"
          >
            Register
          </a>
        </div>
      </form>
    </div>
  </Dialog>
</div>
`;

exports[`Signin component should show error in signin 1`] = `
<div>
  <Dialog
    actions={
      Array [
        <FlatButton
          disabled={false}
          fullWidth={false}
          label="Cancel"
          labelPosition="after"
          labelStyle={Object {}}
          onClick={[MockFunction]}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          primary={false}
          secondary={true}
        />,
      ]
    }
    autoDetectWindowHeight={true}
    autoScrollBodyContent={true}
    modal={true}
    open={true}
    repositionOnUpdate={true}
    title="Login into your account"
  >
    <div
      className="row"
    >
      <form
        className="col s12"
      >
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <Field
              component={[Function]}
              label="Username"
              name="username"
              type="text"
              validate={
                Array [
                  [Function],
                  [Function],
                ]
              }
            />
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <Field
              component={[Function]}
              label="Password"
              name="password"
              type="password"
              validate={[Function]}
            />
          </div>
        </div>
        <ErrorMessage
          message="Error in signin"
        />
        <div
          className="row"
        />
        <div
          className="center-align"
        />
        <div
          className="row center-align"
        >
          <button
            className="btn btn-large waves-effect waves-light
                  indigo accent-2"
            disabled={false}
            id="signin"
            type="submit"
          >
            Sign In
          </button>
        </div>
        <div
          className="row center-align"
        >
          Don't have an account? 
          <a
            className="cursor-pointer blue-text"
            onClick={[MockFunction]}
            role="button"
            tabIndex="0"
          >
            Register
          </a>
        </div>
      </form>
    </div>
  </Dialog>
</div>
`;

exports[`Signin component should show spinner when submitting details 1`] = `
<div>
  <Dialog
    actions={
      Array [
        <FlatButton
          disabled={false}
          fullWidth={false}
          label="Cancel"
          labelPosition="after"
          labelStyle={Object {}}
          onClick={[MockFunction]}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          primary={false}
          secondary={true}
        />,
      ]
    }
    autoDetectWindowHeight={true}
    autoScrollBodyContent={true}
    modal={true}
    open={true}
    repositionOnUpdate={true}
    title="Login into your account"
  >
    <div
      className="row"
    >
      <form
        className="col s12"
      >
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <Field
              component={[Function]}
              label="Username"
              name="username"
              type="text"
              validate={
                Array [
                  [Function],
                  [Function],
                ]
              }
            />
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <Field
              component={[Function]}
              label="Password"
              name="password"
              type="password"
              validate={[Function]}
            />
          </div>
        </div>
        <div
          className="row"
        />
        <div
          className="center-align"
        >
          <Spinner
            size="small"
          />
        </div>
        <div
          className="row center-align"
        >
          <button
            className="btn btn-large waves-effect waves-light
                  indigo accent-2"
            disabled={true}
            id="signin"
            type="submit"
          >
            Sign In
          </button>
        </div>
        <div
          className="row center-align"
        >
          Don't have an account? 
          <a
            className="cursor-pointer blue-text"
            onClick={[MockFunction]}
            role="button"
            tabIndex="0"
          >
            Register
          </a>
        </div>
      </form>
    </div>
  </Dialog>
</div>
`;
